rktl_sim/nodes/simulator

```diff
@@ -358,5 +449,244 @@ class Simulator(object):
+    - generates random max speed and steering throw by calling other functions
+    - used to generate these properties for each car
+    Create a list for each car depending on its properties:
+        Car id
+        Car pose publisher
+        Car odom publisher
+        Car effort subscribers
+        Car command subscribers
+    Publisher:
+        match_status
+    Service:
+        Sim_reset
+        Sim_create_car
+        sim_delete_Car
+    @return the generic car properties that are created for thsi specific run in a dictionary
+    """
+    # general paramters that are the same for every single car
+    # if too constrained in car length between what we detect and what it is - might learn differently
+    # use the get_sim_param in each function case: makes the most sense
+    car_properties = {'length': rospy.get_sim_param('/cars/length'),
+                      'max_speed': self.get_sim_param("/cars/throttle/max_speed"),
+                      'steering_throw': self.get_sim_param("/cars/throttle/max_steering_throw"),
+                      'throttle_tau': rospy.get_sim_param("/cars/throttle/tau"),
+                      'steering_rate': rospy.get_sim_param("/cars/steering/rate"),
+                      'simulate_effort': (self.mode == SimulatorMode.REALISTIC)}
+    self.car_properties = car_properties
+    return car_properties
+
+
+def delete_all_cars(self):
+    car_configs = rospy.get_param('~cars', [])
+    self.reset_lock.acquire()
+    for car_config in car_configs:
+
+        if 'name' not in car_config:
+            rospy.signal_shutdown('no "name" set for car config in sim')
+        car_name = car_config['name']
+        if 'randomize_pose' in car_config:
+            init_pose = None
+
+        self.sim.delete_car(self.car_ids[car_name])
+
+        # Delete all of hte cars subscribers
+        if car_name in self.car_effort_subs:
+            self.car_effort_subs[car_name].unregister()
+            del self.car_effort_subs[car_name]
+        if car_name in self.car_cmd_subs:
+            self.car_cmd_subs[car_name].unregister()
+            del self.car_cmd_subs[car_name]
+        # delete all of the cars publishers
+        if car_name in self.car_odom_pubs:
+            self.car_odom_pubs[car_name].unregister()
+            del self.car_odom_pubs[car_name]
+        if car_name in self.car_pose_pubs:
+            self.car_pose_pubs[car_name].unregister()
+            del self.car_pose_pubs[car_name]
+    # reseting vars to be safe
+    self.car_ids = {}
+    self.car_pose_pubs = {}
+    self.car_odom_pubs = {}
+    self.car_effort_subs = {}
+    self.car_cmd_subs = {}
+    self.reset_lock.release()
+
+    return DeleteCarResponse(True)
+
+
+def generate_all_cars(self):
+    car_configs = rospy.get_param('~cars', [])
+
+    for car_config in car_configs:
+        init_pose = None
+        if 'init_pose' in car_config:
+            init_pose = car_config['init_pose']
```

Comment:
We should be using `get_sim_param` for this

Reply:
done! I also updated the get_sim_param and get_sim_dict function with another variable that specifies if you are fine with receiving None as a return or having an error thrown.